### PR TITLE
Upgrade Tool - Enhanced Logging, and Improved Error Handling

### LINKED ISSUE #21750: Upgrade tool does not provide a useful error message if you are already on the latest version
### Required System information

- Node.js version: v20
- NPM version: Yarn v1
- Strapi version: 5.0.4
- Database: SQLite
- Operating system: Linux Mint 21.2
- Is your project Javascript or Typescript: Javascript

### Describe the bug

When running the `npx @strapi/update@latest minor` if you are already on the latest version does not provide a useful error message.

### Steps to reproduce the behavior

1. Create a new application on the latest version
2. Attempt to run upgrade command using `major`, `minor`, or `patch`
3. See error

### Expected behavior

A useful error message should be given that they are already on the current latest version for their selection

### Screenshots

image

### Code snippets

N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

- Adds a debug log statement to log the application and Strapi versions at the beginning of the upgrade process.
- Includes additional error handling for missing NPM version candidates and corrects minor log typos.
- Display information about the project being already up to date if necessary

### Why is it needed?

QoL improvements.

### How to test it?

1. Run the upgrade command with different tasks (`to`, `patch`, `minor`, `major`) in a project with different versions
2. Make sure the logs display correct and informative messages

### Related issue(s)/PR(s)

fix #21750

### CODE REVIEW

**Comment by Convly** [comment]
Experimental: `0.0.0-experimental.46fff2fd7a1218580f8f7852e7e33574f2082fed`

File: packages/utils/upgrade/src/tasks/upgrade/requirements/major.ts [hunk 1]
```diff
@@ -18,15 +19,18 @@ export const REQUIRE_LATEST_FOR_CURRENT_MAJOR = requirementFactory(
...
+      (match) => semVerFactory(match.version).major === currentMajor
+    );
+
+    if (invalidMatches.length > 0) {
+      const invalidVersions = invalidMatches.map((match) => match.version);
+      const invalidVersionsCount = invalidVersions.length;
 
       throw new Error(
-        `Doing a major upgrade requires to be on the latest v${currentMajor} version, but found ${nbInvalidVersions} versions between the current one and ${target}: ${invalidVersionsAsSemVer}`
+        `Doing a major upgrade requires to be on the latest v${currentMajor} version, but found ${invalidVersionsCount} versions between the current one and ${target}: ${invalidVersions}`
```
Comment (innerdvations) [inline packages/utils/upgrade/src/tasks/upgrade/requirements/major.ts:33]:
```suggestion
        `Doing a major upgrade requires to be on the latest v${currentMajor} version, but found ${invalidVersionsCount} versions between the current one and ${target}. Please upgrade to ${invalidVersions[invalidVersionsCount-1]} and try again.`
```

**Review by innerdvations — APPROVED** [review]
LGTM, just a tiny suggestion

File: packages/utils/upgrade/src/tasks/upgrade/requirements/major.ts (hunk 1 above)
Comment (Convly) [reply packages/utils/upgrade/src/tasks/upgrade/requirements/major.ts:33]: Modifying your suggestion a tiny bit, but thanks!

```suggestion
        `Doing a major upgrade requires to be on the latest v${currentMajor} version, but found ${invalidVersionsCount} versions between the current one and ${target}. Please upgrade to ${invalidVersions.at(-1)} and try again.`
```

**Pull request merged**